netutils/config/parser.py

```diff
@@ -1375,3 +1375,139 @@ class NetironConfigParser(BaseSpaceConfigParser):
     def banner_end(self) -> str:
         """Demarcate End of Banner char(s)."""
         raise NotImplementedError("Extreme Netiron platform doesn't have a banner.")
+
+
+class FastironConfigParser(BaseSpaceConfigParser):
+    """Ruckus FastIron ICX config parser."""
+
+    comment_chars: t.List[str] = ["!"]
+    banner_start: t.List[str] = ["banner"]
+    regex_banner = re.compile(r"^banner\s+(?P<banner_delimiter>\S)")
+
+    def __init__(self, config: str):
+        """Create ConfigParser Object.
+
+        Args:
+            config (str): The config text to parse.
+        """
+        self.delimiter = ""
+        super(FastironConfigParser, self).__init__(config)
+
+    def set_delimiter(self, config_line: str) -> None:
```

Comment:
If previous comment about subclassing doesn't fit, this should be a `property.setter` 
example
```python
    @banner_end.setter
    def banner_end(self, banner_start_line: str) -> None:
        banner_parsed = self.regex_banner.match(banner_start_line)
        if not banner_parsed:
            raise ValueError("There was an error parsing your banner, the end of the banner could not be found")
        self._banner_end = banner_parsed.groupdict()["banner_delimiter"]
```